scripts/publish.rs

```diff
@@ -0,0 +1,127 @@
+    let cargo_pkgid_str = str::from_utf8(&cargo_pkgid_output.stdout)
+        .expect("Failed to convert pkgid output into a str");
+
+    // Extract only the local crate version from str
+    let (_, local_version) = cargo_pkgid_str
+        .split_once('#')
+        .expect("Failed to get local crate version");
+
+    local_version.trim_end().to_string()
+}
+
+// Obtain remote crate version
+fn remote_version(crate_name: &str) -> String {
+    // Obtain remote crate version contained in cargo search data
+    let cargo_search_output = Command::new("cargo")
+        .args(["search", crate_name, "--limit", "1"])
+        .output()
+        .expect("Failed to run cargo search");
+
+    // Convert cargo search output into a str
+    let remote_version_str = str::from_utf8(&cargo_search_output.stdout)
+        .expect("Failed to convert cargo search output into a str");
+
+    // Extract only the remote crate version from str
+    let (remote_version, _) = remote_version_str
+        .split_once('=')
+        .map(|(_, second)| second.trim_start().split_once(' '))
+        .flatten()
+        .expect("Failed to get remote crate version");
+
+    remote_version.trim_matches('"').to_string()
+}
+
+// If stdout is not empty, write it, otherwise
+// write stderr and exit with code error 1
+fn stdout_and_stderr_write(output: Output, message_stdout: &str, message_stderr: &str) {
+    if !output.stdout.is_empty() {
+        io::stdout()
+            .write_all(&output.stdout)
+            .expect(message_stdout);
+    }
+
+    if !output.stderr.is_empty() {
+        io::stderr()
+            .write_all(&output.stderr)
+            .expect(message_stderr);
+        std::process::exit(1);
+    }
+}
+
+// Publishes a crate
+fn publish(crate_name: String) {
+    // Run cargo publish --dry-run
+    let cargo_publish_dry = Command::new("cargo")
+        .args(["publish", "-p", &crate_name, "--dry-run"])
+        .output()
+        .expect("Failed to run cargo publish --dry-run");
+
+    // Write cargo publish --dry-run either on stdout or on stderr
+    stdout_and_stderr_write(
+        cargo_publish_dry,
+        "Failed to write cargo publish --dry-run on stdout",
+        "Failed to write cargo publish --dry-run on stderr",
+    );
+
+    let crates_io_token =
+        env::var(CRATES_IO_API_TOKEN).expect("Failed to retrieve the crates.io API token");
+
+    // Publish crate
+    let cargo_publish = Command::new("cargo")
+        .args(["publish", "-p", &crate_name, "--token", &crates_io_token])
+        .output()
+        .expect("Failed to run cargo publish");
+
+    // Write cargo publish either on stdout or on stderr
+    stdout_and_stderr_write(
+        cargo_publish,
+        "Failed to write cargo publish on stdout",
+        "Failed to write cargo publish on stderr",
+    );
+}
+
+fn main() {
+    // Get crate name
+    let crate_name = env::args()
+        .nth(1)
+        .expect("You need to pass the crate name as first argument!");
```

Comment:
What is the 0 th argument?

Reply:
The executable path, so typically you'd skip the 0th arg.